splitgraph/ingestion/athena/__init__.py

```diff
@@ -0,0 +1,136 @@
+        },
+    }
+
+    params_schema = {
+        "type": "object",
+        "properties": {
+            "region_name": {
+                "type": "string",
+                "title": "S3 region",
+                "description": "Region of the S3 bucket",
+            },
+            "schema_name": {
+                "type": "string",
+                "title": "Schema",
+                "description": "Athena database name",
+            },
+            "s3_staging_dir": {
+                "title": "S3 results folder",
+                "type": "string",
+                "description": "Folder for storing query output",
+            },
+        },
+        "required": ["region_name", "schema_name", "s3_staging_dir"],
+    }
+
+    supports_mount = True
+    supports_load = True
+    supports_sync = False
+
+    commandline_help = """Mount a Amazon Athena database.
+
+This will mount an Athena schema or a table:
+
+\b
+```
+$ sgr mount athena s3 -o@- <<EOF
+{
+    "aws_access_key_id": "ABCD",
+    "aws_secret_access_key": "abcd",
+    "region_name": "eu-west-3",
+    "schema_name": "mydatabase",
+    "s3_staging_dir": "s3://my-bucket/output/",
+}
+EOF
+```
+    """
+
+    commandline_kwargs_help: str = (
+        build_commandline_help(credentials_schema) + "\n" + build_commandline_help(params_schema)
+    )
+
+    _icon_file = "snowflake.svg"
+
+    def __init__(
+        self,
+        engine: "PostgresEngine",
+        credentials: Credentials,
+        params: Params,
+        tables: Optional[TableInfo] = None,
+    ):
+        super().__init__(engine, credentials, params, tables)
+
+    def get_fdw_name(self):
+        return "multicorn"
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "Amazon Athena"
+
+    @classmethod
+    def get_description(cls) -> str:
+        return "Query data in Amazon S3 files and folders"
+
+    def get_table_options(
+        self, table_name: str, tables: Optional[TableInfo] = None
+    ) -> Dict[str, str]:
+        result = super().get_table_options(table_name, tables)
+        result["tablename"] = result.get("tablename", table_name)
+        return result
+
+    def get_server_options(self):
+        options: Dict[str, Optional[str]] = {
+            "wrapper": "multicorn.sqlalchemyfdw.SqlAlchemyFdw",
+            "db_url": self._build_db_url(),
+        }
+
+        # For some reason, in SQLAlchemy, if this is not passed
+        # to the FDW params (even if it is in the DB URL), it doesn't
+        # schema-qualify tables and server-side cursors don't work for scanning
+        # (loads the whole table instead of scrolling through it).
+        if "schema" in self.params:
+            options["schema"] = self.params["schema"]
+
+        return options
+
+    def _build_db_url(self) -> str:
+        """Construct the SQLAlchemy Amazon Athena db_url"""
+
+        aws_access_key_id = self.params["aws_access_key_id"]
+        aws_secret_access_key = self.params["aws_secret_access_key"]
```

Comment:
I think this comes from credentials (not merged into params)

```suggestion
        aws_access_key_id = self.credentials["aws_access_key_id"]
        aws_secret_access_key = self.credentials["aws_secret_access_key"]
```

Reply:
Curiously, it actually works properly either way.